### PR TITLE
Merge and stack

### PR DESCRIPTION
**Pull request by lilyclements**
Altered the TestOKEnabled codes

### CODE REVIEW

File: instat/dlgStack.vb [hunk 1]
```diff
@@ -110,7 +113,7 @@ Public Class dlgStack
     End Sub
 
     Private Sub ucrReceiverColumnsToBeStack_SelectionChanged() Handles ucrReceiverColumnsToBeStack.SelectionChanged
-        If Not ucrReceiverColumnsToBeStack.IsEmpty Then
+        If Not ucrReceiverColumnsToBeStack.IsEmpty = False Then
```
Comment (dannyparsons) [inline instat/dlgStack.vb]: I think you don't need the = False here.

File: instat/dlgStack.vb [hunk 2]
```diff
@@ -162,9 +167,19 @@ Public Class dlgStack
...
         SetStackIntoText()
     End Sub
+
+    Private Sub ucrIDVariablesReceiver_SelectionChanged(sender As Object, e As EventArgs) Handles ucrColumnsToCarryReceiver.SelectionChanged
+
+    End Sub
+
+    Private Sub ucrReceiverColumnsToBeStack_SelectionChanged(sender As Object, e As EventArgs) Handles ucrReceiverColumnsToBeStack.SelectionChanged
+
+    End Sub
```
Comment (dannyparsons) [inline instat/dlgStack.vb]: You can delete these two empty subs

File: instat/dlgMerge.vb [hunk 3]
```diff
@@ -59,18 +59,20 @@ Public Class dlgMerge
...
+            ucrBase.OKEnabled(False)
+        Else
+            If rdoChooseMergeColumns.Checked Then
+                If (Not ucrReceiverFirstDF.IsEmpty() AndAlso Not ucrReceiverSecondDF.IsEmpty) OrElse lstKeyColumns.Items.Count > 0 Then
+                    ucrBase.OKEnabled(True)
+                Else
+                    ucrBase.OKEnabled(False)
+                End If
+            End If
+            If rdoMergeByAllColumns.Checked Then
```
Comment (dannyparsons) [inline instat/dlgMerge.vb]: I think this If should be an Else If of the one above.

**Comment by lilyclements** [comment]
Thank you! The =False would have snuck in when I changed the code and then had to switch it back to the initial code! Sorry about the extra subs too, I keep accidentally double clicking.

If I make for Merge it as ElseIf then this occurs
    Private Sub TestOKEnabled()        If ucrNewDataFrameName.IsEmpty() Then            ucrBase.OKEnabled(False)        ElseIf rdoChooseMergeColumns.Checked Then            If (Not ucrReceiverFirstDF.IsEmpty() AndAlso Not ucrReceiverSecondDF.IsEmpty) OrElse lstKeyColumns.Items.Count > 0 Then                    ucrBase.OKEnabled(True)                Else                    ucrBase.OKEnabled(False)                End If                If rdoMergeByAllColumns.Checked Then                    ucrBase.OKEnabled(True)                End If            End If
And so rdoMergeByAllColumns.Checked becomes part or the rdoChooseMergeColumns.Checked 'umbrella' and the code doesn't work. Did you mean to put 'ElseIf' in a different place?

Date: Mon, 19 Sep 2016 05:10:14 -0700

**Pull request closed without merging**